src/commands/mod.rs

```diff
@@ -684,23 +684,43 @@ struct ResolveArgs {
 /// as they had in the source (`--from`). This is typically used for undoing
 /// changes to some paths in the working copy (`jj restore <paths>`).
 ///
+/// If only one of `--from` or `--to` is specified, the other one defaults to
+/// the working copy.
+///
 /// When neither `--from` nor `--to` is specified, the command restores into the
-/// working copy from its parent. If one of `--from` or `--to` is specified, the
-/// other one defaults to the working copy.
+/// working copy from its parent(s). `jj restore` without arguments is similar
+/// to `jj abandon`, except that it leaves an empty revision with its
+/// description and other metadata preserved.
 ///
 /// See `jj diffedit` if you'd like to restore portions of files rather than
 /// entire files.
 #[derive(clap::Args, Clone, Debug)]
 struct RestoreArgs {
+    /// Restore only these paths (instead of all paths)
+    #[arg(value_hint = clap::ValueHint::AnyPath)]
+    paths: Vec<String>,
     /// Revision to restore from (source)
     #[arg(long)]
     from: Option<RevisionArg>,
     /// Revision to restore into (destination)
     #[arg(long)]
     to: Option<RevisionArg>,
-    /// Restore only these paths (instead of all paths)
-    #[arg(value_hint = clap::ValueHint::AnyPath)]
-    paths: Vec<String>,
+    /// Undo the changes in a revision as compared to the merge of its parents.
+    ///
+    /// This undoes the changes that can be seen with `jj diff -r REVISION`. If
+    /// `REVISION` only has a single parent, this option is equivalent to `jj
+    ///  restore --to REVISION --from REVISION-`.
+    ///
+    /// The default behavior of `jj restore` is equivalent to `jj restore
+    /// --revision-from-parents @`.
+    //
+    // If we followed the pattern of `jj diff` and `jj diffedit`, this option
+    // could simply be called `--revision`/`-r`. However, that would make it
+    // likely that someone unfamiliar with this pattern would use `-r` when they
+    // wanted `--from`. This would make a different revision empty, and the user
+    // might not even realize something went wrong.
+    #[arg(long, visible_alias="rp", value_name="REVISION", conflicts_with_all=["to", "from"])]
+    revision_from_parents: Option<RevisionArg>,
```

Comment:
Do you think we should call it `--changes-in/-c`? I think that sounds good whether or not we rename `diff -r` to `diff -c` (but especially if we do, of course). Feel free to merge this as if you prefer to keep the current name.

Reply:
FWIW, I would personally probably prefer simply using `--revision/-r` instead of `--revision-from-parents`, which would match the current behavior of `jj diff`. But it sounds like others would find that confusing.

Reply:
Indeed, I'm currently planning to change it to `--changes-in`.

I'm also thinking of adding a hidden `-r` option that prints a hint telling the user to use `--from` or `-c`. This will probably go into this PR, but might go in a separate one.

Thank you!

Reply:
Done. I'll wait a few hours before merging in case anyone has any comments.